Add more smart pointers to CheckboxInputType.cpp

rwlbuis opened the pull request:
#### 2fa2e804cc1bd7c1528e35b071c5943ac62094eb
<pre>
Add more smart pointers to CheckboxInputType.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=272897">https://bugs.webkit.org/show_bug.cgi?id=272897</a>

Reviewed by Chris Dumez.

Add more smart pointers to CheckboxInputType.cpp.

* Source/WebCore/html/CheckboxInputType.cpp:
(WebCore::CheckboxInputType::handleMouseDownEvent):
(WebCore::CheckboxInputType::handleMouseMoveEvent):
(WebCore::CheckboxInputType::handleTouchEvent):
(WebCore::CheckboxInputType::willDispatchClick):
(WebCore::CheckboxInputType::didDispatchClick):
(WebCore::CheckboxInputType::disabledStateChanged):

Canonical link: <a href="https://commits.webkit.org/277810@main">https://commits.webkit.org/277810@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/38b186195c360810ba51537b1f253608d9b36954

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/48660 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/27871 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/51621 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/51347 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/44725 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/50965 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/33808 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/25401 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/39790 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/49242 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/25532 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/41988 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/20887 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/23010 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/43166 "Found 1 new test failure: fast/css/viewport-height-border.html (failure)") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/6716 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/44948 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/43652 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/53255 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/23707 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/20006 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/47087 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/24973 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/42193 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/46017 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10720 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/25777 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/24694 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/html/CheckboxInputType.cpp:
```diff
@@ -104,23 +104,25 @@ void CheckboxInputType::handleKeyupEvent(KeyboardEvent& event)
+    ASSERT(element);
 
-    if (!event.isTrusted() || !isSwitch() || element()->isDisabledFormControl() || !element()->renderer())
+    if (!event.isTrusted() || !isSwitch() || element->isDisabledFormControl() || !element->renderer())
```
Maybe we should split into 2 if checks so we don't Ref the element unnecessarily in the early return case:
```
if (!event.isTrusted() || !isSwitch())
    return;

RefPtr element = this->element();
ASSERT(element);
if (element->isDisabledFormControl() || !element->renderer()))
    return;
```

rwlbuis replied:
Good idea, done.

cdumez commented on Source/WebCore/html/CheckboxInputType.cpp:
```diff
@@ -104,23 +104,25 @@ void CheckboxInputType::handleKeyupEvent(KeyboardEvent& event)
 void CheckboxInputType::handleMouseMoveEvent(MouseEvent& event)
 {
-    ASSERT(element());
+    RefPtr element = this->element();
```
This early variable can be moved *after* the `!isSwitchPointerTracking()` early return.

rwlbuis replied:
Sure, done.

cdumez commented on Source/WebCore/html/CheckboxInputType.cpp:
```diff
@@ -225,10 +229,11 @@ void CheckboxInputType::willDispatchClick(InputElementClickState& state)
 
 void CheckboxInputType::didDispatchClick(Event& event, const InputElementClickState& state)
 {
+    RefPtr element = this->element();
```
Should be moved inside the `if (event.defaultPrevented() || event.defaultHandled()) {` scope.

rwlbuis replied:
Fixed.

cdumez commented on Source/WebCore/html/CheckboxInputType.cpp:
```diff
@@ -281,8 +286,9 @@ bool CheckboxInputType::matchesIndeterminatePseudoClass() const
-    if (isSwitch() && element()->isDisabledFormControl()) {
+    RefPtr element = this->element();
+    ASSERT(element);
+    if (isSwitch() && element->isDisabledFormControl()) {
```
I think we should split this `if` case into 2 so we don't need to ref the element if `isSwitch()` is false, which is likely common.

rwlbuis replied:
Done.

cdumez reviewed (approved):
LGTM with fixes.

rwlbuis committed:
Add more smart pointers to CheckboxInputType.cpp
https://bugs.webkit.org/show_bug.cgi?id=272897

Reviewed by Chris Dumez.

Add more smart pointers to CheckboxInputType.cpp.

* Source/WebCore/html/CheckboxInputType.cpp:
(WebCore::CheckboxInputType::handleMouseDownEvent):
(WebCore::CheckboxInputType::handleMouseMoveEvent):
(WebCore::CheckboxInputType::handleTouchEvent):
(WebCore::CheckboxInputType::willDispatchClick):
(WebCore::CheckboxInputType::didDispatchClick):
(WebCore::CheckboxInputType::disabledStateChanged):

Canonical link: https://commits.webkit.org/277810@main
